R/zchunk_L100.Population_downscale_ctry.R

```diff
@@ -34,19 +34,129 @@ module_socioeconomics_L100.Population_downscale_ctry_DISABLED <- function(comman
     UN_popTot <- get_data(all_data, "socioeconomics/UN_popTot")
 
     # ===================================================
-    # TRANSLATED PROCESSING CODE GOES HERE...
-    #
-    # If you find a mistake/thing to update in the old code and
-    # fixing it will change the output data, causing the tests to fail,
-    # (i) open an issue on GitHub, (ii) consult with colleagues, and
-    # then (iii) code a fix:
-    #
-    # if(OLD_DATA_SYSTEM_BEHAVIOR) {
-    #   ... code that replicates old, incorrect behavior
-    # } else {
-    #   ... new code with a fix
-    # }
-    #
+
+    ## (1) Historical population by country
+
+    # First clean up Maddison raw data -- NOTE: Maddison data are used to develop population ratios relative to 1950 to combine with UN data from 1950 onward
+    pop_thous_ctry_reg <- Maddison_population %>%
+      filter(!is.na(Country)) %>%  # Remove values with missing country name (all year values are also missing for these--probably just blank rows in the original data)
+      rename(Maddison_ctry = Country,
+             pop = value) %>%  # Change name to match socioeconomics_ctry mapping file
+      left_join(socioeconomics_ctry, by = "Maddison_ctry")  # Join with iso codes
+
+    # Second, estimate population values prior to 1950 for countries in aggregate regions. This is what we want: pop_country_t = (pop_aggregate_t / pop_aggregate_1950) * pop_country_1950
+    # Generate a scalar for population in each aggregate region in 1950 (to generate the population ratios)
+    agg_ratio <- pop_thous_ctry_reg %>%
+      select(Maddison_ctry, year, pop) %>%
+      filter(year <= socioeconomics.AGG_BASE_YEAR & (Maddison_ctry == "Total Former USSR" | Maddison_ctry == "Czechoslovakia" | Maddison_ctry == "Yugoslavia")) %>%  # Only want years prior to 1951 for the three regions
+      group_by(Maddison_ctry) %>%  # Group to perform action on each aggregate region individually
+      mutate(ratio = pop/pop[year == socioeconomics.AGG_BASE_YEAR]) %>%  # Create ratio of population in prior years to population in 1950
```

Comment:
I'm a little surprised that this works, since `pop` and `pop[year==...]` have different lengths.  I suppose it's because within each grouping the year criterion selects exactly row, and you're allowed to divide a vector by a scalar.